surveys/2024-04/questions.md

```diff
@@ -0,0 +1,142 @@
+    - With a Node.js version manager: nvm, n, nave, nvs, volta, etc.
+    - Using the official installer
+    - Using package managers like npm or yarn and install Node.js as a package
+    - Using a system package manager: apt-get, brew, dnf, yum …
+    - Downloading binaries directly from Nodejs.org
+    - Building Node.js from Source
+    - Other
+
+- What package manager do you use ? (multiple choice - Other should give an input)
+    - npm
+    - yarn
+    - cnpm
+    - pnpm
+    - Other ? 
+
+- If you use a version manager which one do you use (multiple choice - Other should give an input)
+    - nvm
+    - n
+    - nave
+    - nvs
+    - volta
+    - Other ?
+
+## Project Priorities and Direction
+
+- Are you aware of the defined Node.js [Technical Priorities](https://github.com/nodejs/node/blob/main/doc/contributing/technical-priorities.md)? (single choice)
+    - Yes
+    - No
+
+- Which of the current technical priorities are important to you ? (multiple choice)
+    - Modern HTTP
+    - Suitable types for end-users
+    - Documentation
+    - WebAssembly
+    - ESM
+    - Support for features from the latest ECMAScript spec
+    - Observability
+    - Permissions/policies/security model
+    - Better multithreaded support
+    - Single Executable Applications
+
+- Are there technical priorities that you believe are missing (open question)
+
+- What is important to you ? (multiple choice)
+    - Good understanding of the direction of the project
+    - Ability to affect the direction of the project
+    - Consumable APIs and docs
+    - Predictable and stable releases
+    - Innovation at a consumable pace
+    - Easy Installation
+    - Easy issue reporting, resolution and collaboration
+    - Broad deployment platform support
+    - Broad desktop platform support
+    - Consistent and intuitive error handling
+    - Runtime diagnostic tooling
+    - Development time diagnostic tooling
+    - Relevant APIs in core
+    - Module/dependency info and management
+    - Ways to fund their work
+    - Ability to assess impact of changes they make
+    - Reasonable resource usage/performance
+    - Good security and CVE practices in the project
+    - Better CVE management in the ecosystem
+    - Good CI infrastructure and experience in the project
+    - Supportive Collaborators and Environment in the project
+    - Better ways to build consensus in the project
+    - Easy contribution workflow
+    - Ability to embed and bundle the Node.js runtime
+    - A well maintained and secure standard library
+    - Assets that show Node.js is a good choice
+
+- What is important to you that is not in this list? (open question)
+
+## Technical Questions
+
+- Are you using native ESM (e.g. `"type": "module"`,  `.mjs` or other solutions) in your Node.js Development?
+    - Yes
+    - No
+
+- Are you planning to convert an application to native ESM or use native ESM for a future application?
+    - Yes
+    - No
+    
```

Comment:
From the summit we also discussed about adding questions about usage of features. From what I can remember:

```suggestion
 - Are you using the following experimental features of Node.js (multiple choice)?
     - Corepack
     - Async Hooks
     - Permission model / Policies
     - Single Executable Application
     - Startup Snapshot
     - Loader hooks (`--loader` or `module.register()`)
     - Network import (`import 'http://...'`)
     - WASI
     - VM Modules (`--experimental-vm-modules`)
     - Watch mode (`--watch`)
     - WebSocket client
     - Trace events (`--trace-event-categories` or `node:trace_events`)
     - Dotenv (`--env-file`)
     - Import attributes

  - Are you using the following new stable features (multiple choice):
   - Test runner (`node:test`)
   - Web Crypto (`globalThis.crypto` or `crypto.webcrypto`)
   - Web Streams (`node:stream/web`)
   - Fetch
```

Reply:
+1 for that because having data about Corepack look important

Reply:
Please don’t mention `--loader`, it will probably be removed. Just `module.register`.

Reply:
This survey is a good opportunity to get some usage data on `--loader`/`--experimental-loader` – if we don't mention it, we can only guess whether removing it makes sense or not.

Reply:
`--loader` isn't documented. It's an accident that it exists. We shouldn't be publicizing it.